node/pkg/watchers/solana/client_test.go

```diff
@@ -6,10 +6,53 @@ import (
 	"github.com/stretchr/testify/assert"
 
 	"github.com/gagliardetto/solana-go"
+	"github.com/gagliardetto/solana-go/rpc"
 )
 
 func TestVerifyConstants(t *testing.T) {
 	// If either of these ever change, message publication and reobservation will break.
 	assert.Equal(t, SolanaAccountLen, solana.PublicKeyLength)
 	assert.Equal(t, SolanaSignatureLen, len(solana.Signature{}))
 }
+
+func TestCheckCommitment(t *testing.T) {
+	type test struct {
+		commitment      string
+		watcher         string
+		isReobservation bool
+		result          bool
+	}
+	tests := []test{
+		// New observation success cases
+		{commitment: "finalized", watcher: "finalized", isReobservation: false, result: true},
+		{commitment: "confirmed", watcher: "confirmed", isReobservation: false, result: true},
```

Comment:
Do you want to add a test case where commitments are the same and isReobs is true?